Added a new route for user.ts for fetching user information

jesreu opened the pull request:
Added a new route in user.ts for building the frontend userlist functionality.

valju reviewed (changes requested):
line 17 says 'knex', but lines 59 and 66 say 'db_knex'

valju commented:
Nice knex studying. There is most likely also a way to do this without the raw part. I wonder whether it would even create the object structure for us. A list of users, each of which might have a list of departments (or departmenPlanner markings).

The second join is just normal (inner) join. DepartmentPlanner is a join table, so there are no lines on left (DepartmentPlanner) that would not have counter part on right (Department). 

But the first one, yes, must be left join, there might be users who are not marked as Department planners at all.

jesreu commented:
Removed the unused knex import on line 17, probably added it as mistake when figuring out the knex.raw part. 

I kept the second join as a left join. Because when testing with an inner join on the local database it wouldn't fetch information for user admin who isn't a planner for any departments. 

Also removed spacing from the wording of the department planner section so it's easier to access in frontend,
